### PR TITLE
replacing DispatchDB instances with datastore factory

### PR DESCRIPTION
**Pull request by scottwn**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by scottwn** [comment]
@FyzHsn @AlejandroEsquivel this is a WIP to replace the `DispatchDB()` instances with the `DataStore` factory as [we discussed on Wednesday](https://agnostiqworkspace.slack.com/archives/C03KZRV93J5/p1658335714149149)
Please feel free to work on this branch, there are still things that need to be updated, but hopefully it's a starting point.

**Comment by wjcunningham7** [comment]
This pull request is stale because it has been in review for at least three days. It will be closed in two more days.

File: covalent_dispatcher/_service/app.py [hunk 1]
```diff
@@ -99,7 +100,7 @@ def get_result(dispatch_id) -> Response:
     wait = args.get("wait", default=False, type=lambda v: v.lower() == "true")
     status_only = args.get("status_only", default=False, type=lambda v: v.lower() == "true")
     while True:
-        with Session(DispatchDB()._get_data_store().engine) as session:
+        with workflow_db.session as session:
```
Comment (FyzHsn) [inline covalent_dispatcher/_service/app.py:103]:
```suggestion
        with workflow_db.session as session:
        with workflow_db.session() as session:
```

**Review by FyzHsn — CHANGES_REQUESTED** [review]

**Comment by FyzHsn** [comment]
@scottwn Found a syntax error - maybe has something to do with the failing tests. But otherwise, looking good!

**Comment by scottwn** [comment]
@FyzHsn @AlejandroEsquivel I'm seeing this error in logs every time I `GET` `/result/<dispatch_id>`.
```
  File "/Users/scott/covalent/covalent_dispatcher/_service/app.py", line 103, in get_result
    with workflow_db.session() as session:
AttributeError: __enter__
```
[Stackoverflow](https://stackoverflow.com/questions/43429942/attributeerror-enter-using-with-statement-sqlalchemy-session) suggests that it's because I'm not calling the function (I am calling it since the syntax fix that @FyzHsn identified) or because of SQLAlchemy `<1.4` (we are on `1.4.37`). Any idea how to fix this?

**Comment by scottwn** [comment]
@AlejandroEsquivel another question, why is `Session` capitalized? shouldn't it be lowercase since it's a class attribute?
https://github.com/AgnostiqHQ/covalent/blob/07f7fb83de078c82d5e1f06f4aab0ae01b868cb2/covalent/_data_store/datastore.py#L54

**Comment by wjcunningham7** [comment]
This pull request is stale because it has been in review for at least three days. It will be closed in two more days.

**Comment by AlejandroEsquivel** [comment]
> @AlejandroEsquivel another question, why is `Session` capitalized? shouldn't it be lowercase since it's a class attribute?
> 
> https://github.com/AgnostiqHQ/covalent/blob/07f7fb83de078c82d5e1f06f4aab0ae01b868cb2/covalent/_data_store/datastore.py#L54

@scottwn this is as as far I understand `Session` is a class, and it also seems to be capitalized in SQLAlchemy documentation so I am capitalizing it for consistency, you can typically import `Session` directly [see here](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#opening-and-closing-a-session), however you always need to specify an `engine` to initialize the session. I am using the `sessionmaker` method to create a new `Session` reference which can be used in the exact same way but without needing to specify an `engine` ([docs on this here](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#using-a-sessionmaker))

**Comment by AlejandroEsquivel** [comment]
> @FyzHsn @AlejandroEsquivel I'm seeing this error in logs every time I `GET` `/result/<dispatch_id>`.
> 
> ```
>   File "/Users/scott/covalent/covalent_dispatcher/_service/app.py", line 103, in get_result
>     with workflow_db.session() as session:
> AttributeError: __enter__
> ```
> 
> [Stackoverflow](https://stackoverflow.com/questions/43429942/attributeerror-enter-using-with-statement-sqlalchemy-session) suggests that it's because I'm not calling the function (I am calling it since the syntax fix that @FyzHsn identified) or because of SQLAlchemy `<1.4` (we are on `1.4.37`). Any idea how to fix this?

@scottwn if there is a flaw in how the context manager is written one thing to try would be using the `Session` class directly, therefore:
https://github.com/AgnostiqHQ/covalent/blob/19661f137e0cb04f1bdb0d5db7ecbabc304628f6/covalent_dispatcher/_service/app.py#L103
could be `with workflow_db.Session() as session:` or `with workflow_db.Session.begin()` as in the docs https://docs.sqlalchemy.org/en/14/orm/session_basics.html#using-a-sessionmaker

**Review by FyzHsn — APPROVED** [review]

File: tests/covalent_dispatcher_tests/_cli/cli_test.py [hunk 2]
```diff
@@ -41,7 +41,6 @@ def test_cli(mocker):
         ("python" in response.output.lower())
         and ("agnostiq" in response.output.lower())
         and ("copyright" in response.output.lower())
-        and (current_version in response.output.lower())
```
Comment (scottwn) [inline tests/covalent_dispatcher_tests/_cli/cli_test.py:44]: do we not want the version in the output?

**Review by FyzHsn — DISMISSED** [review]

File: tests/covalent_dispatcher_tests/_cli/cli_test.py (hunk 2 above)
Comment (FyzHsn) [reply tests/covalent_dispatcher_tests/_cli/cli_test.py:44]: The actual version in the branch is ahead (0.158.0) vs the version in the CLI --version command (Release version 0.156.0). So, I figured this test logic is not quite right.

**Review by santoshkumarradha — APPROVED** [review]
Docs are perfect ! 🚀

**Review by FyzHsn — APPROVED** [review]

**Review by santoshkumarradha — DISMISSED** [review]
Docs are gtg!!

**Comment by scottwn** [comment]
@cjao @FyzHsn this is what's failing in the tests
```
>           assert (
                load_file(storage_path=lattice_storage_path, filename=lattice_row.executor_filename)
                == "dask"
            )
E           AttributeError: 'Lattice' object has no attribute 'executor_filename'
```
Was `executor_filename` removed in v11 (should the test be changed)? Or is the test correct and I have to fix the lattice table?

**Comment by cjao** [comment]
Hi @scottwn, the test should be [updated](https://github.com/AgnostiqHQ/covalent/commit/e81944d696ba34d3b00eccfc14bf496c5f10bdd8#diff-16b551c864dae9dcd147a9531878025fd164c59b699bb00dc4cea1a3d582316eR63). V11 split `executor_filename` into two fields, `executor` and `executor_data_filename`.

**Review by FyzHsn — DISMISSED** [review]

**Review by FyzHsn — APPROVED** [review]

**Review by santoshkumarradha — APPROVED** [review]

**Pull request merged**